### PR TITLE
Make e164 optional in ServiceAddress

### PR DESCRIPTION
**Pull request by gferon**
This is a necessary step for future improvements, like #27.

### CODE REVIEW

File: libsignal-service/src/sender.rs [hunk 1]
```diff
@@ -139,12 +139,14 @@ where
                             ),
                         )?;
                     }
-                    self.cipher.store_context.delete_session(
-                        &libsignal_protocol::Address::new(
-                            &recipient.e164,
-                            *extra_device_id,
-                        ),
-                    )?;
+                    if let Some(ref e164) = recipient.e164 {
```
Comment (rubdos) [inline libsignal-service/src/sender.rs:142]: Sessions with Uuid should be deleted too, no? Maybe `get_preferred_protocol_address`?

**Review by rubdos — CHANGES_REQUESTED** [review]

File: libsignal-service/src/sender.rs (hunk 1 above)
Comment (gferon) [reply libsignal-service/src/sender.rs:142]: Yes, it was already done before? (open the code above).

Comment (rubdos) [reply libsignal-service/src/sender.rs:142]: Yep, missed that, sorry for that.

**Review by rubdos — APPROVED** [review]
LGTM

**Pull request merged**